src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,279 @@
+        document.body.appendChild(iframe)
+        this.scrollToTopHack()
+        this.attachMouseListeners(iframe)
+        return iframe
+    }
+
+    private attachMouseListeners(iframe: HTMLIFrameElement) {
+        this.attachMouseOverListener(iframe)
+        this.attachMouseOutListener(iframe)
+    }
+
+    private attachMouseOverListener(iframe: HTMLIFrameElement) {
+        document.addEventListener('mouseover', (e: Event) => {
+            const target = e.target as HTMLElement
+            const isPageRef = this.isTargetPageRef(target)
+            if (isPageRef) {
+                const text = this.getTargetInnerText(target)
+                this.hoveredElement = target
+                const url = Navigation.getPageUrlByName(text)
+                if ((!this.getIFrameByUrl(url) || !this.getIsIFrameVisibleByUrl(url)) && iframe) {
+                    iframe = this.prepIframeForDisplay(iframe, url)
+                }
+                this.setTimerForPopup(iframe, target)
+            }
+        })
+    }
+
+    private attachMouseOutListener(iframe: HTMLIFrameElement) {
+        document.addEventListener('mouseout', (e: MouseEvent) => {
+            const target = e.target as HTMLElement
+            const relatedTarget = e.relatedTarget as HTMLElement
+            if (
+                this.isHoveredOutFromTarget(target, relatedTarget, iframe) ||
+                this.isHoveredOutFromIframe(target, relatedTarget, iframe) ||
+                !this.isHoveredElementPresentInBody()
+            ) {
+                this.hoveredElement = null
+                this.clearPopupTimeout()
+                this.resetIframeForNextHover(iframe)
+                this.destroyPopper()
+            }
+        })
+    }
+
+    private setTimerForPopup(iframe: HTMLIFrameElement, target: HTMLElement) {
+        if (!this.popupTimeout) {
+            this.popupTimeout = window.setTimeout(() => {
+                if (iframe) {
+                    iframe = this.showIframe(iframe)
+                    this.makePopper(target, iframe)
+                }
+            }, this.popupTimeoutDuration)
+        }
+    }
+
+    private getTargetInnerText(target: HTMLElement) {
+        // remove '#' for page tags
+        const isPageRefTag = this.isTargetPageRefTag(target)
+        return isPageRefTag ? target.innerText.slice(1) : target.innerText
+    }
+
+    private isTargetPageRefTag(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref-tag')
+    }
+
+    private isTargetPageRef(target: HTMLElement) {
+        return target.classList.contains('rm-page-ref')
+    }
+
+    private destroyPopper() {
+        if (this.popper) {
+            this.popper.destroy()
+            this.popper = null
+        }
+    }
+
+    private resetIframeForNextHover(iframe: HTMLIFrameElement) {
+        if (iframe) {
+            if (iframe.contentDocument) {
+                // scroll to top when removed, so the next popup is not scrolled
```

Comment:
should this be using `scrollToTopHack` ?

Reply:
this is a different scollToTop hack :)

Reply:
why do they have to be different. would scrolling to the top of html not be enough here too?

Reply:
@Stvad that is during loading. When a post is loaded, you don't scroll on the `html`, you scroll on a div inside `.roam-center`. Need to set the scrolltop of that.